changes/watch_darwin.go

```diff
@@ -0,0 +1,84 @@
+// heavily inspired by https://github.com/fsnotify/go.fsevents
```

Comment:
I hope it's not too heavily inspired by go.fsevents, because that code doesn't have an open source license. See https://github.com/fsnotify/go.fsevents/pull/7#issuecomment-63198985. 

Inspiration from https://github.com/fsnotify/fsevents would be better :-)